### PR TITLE
Improve support for editing QML source giles on the cerbo

### PR DESCRIPTION
**Pull request by DanielMcInnes**
I think this now works for all QML source files, except for Main.qml - this is loaded from .cpp, and needs to use the resource system. We could move most stuff out of Main.qml and into a new file if we think it's necessary.

### CODE REVIEW

File: ApplicationContent.qml [hunk 1]
```diff
@@ -88,16 +88,27 @@ Item {
...
 		if (Qt.platform.os !== "wasm") {
-			_inputComponent = Qt.createComponent(Global.appUrl("/components/InputPanel.qml"), Component.Asynchronous)
-			_inputComponent.statusChanged.connect(function() {
-				if (_inputComponent.status === Component.Ready) {
-					Global.inputPanel = _inputComponent.createObject(root, { mainViewItem: mainView })
-				} else if (_inputComponent.status === Component.Error) {
-					console.warn("Cannot load InputPanel:", _inputComponent.errorString())
-				}
-			})
+			loader.sourceComponent = inputPanelComponent
```
Comment (jpetrell) [inline ApplicationContent.qml:111]: Could just write

```
Loader {
    active: Qt.platform.os !== "wasm"
    asynchronous: true
    sourceComponent: InputPanel { .. }
}
```

File: components/PageStack.qml [hunk 2]
```diff
@@ -69,7 +69,7 @@ C.StackView {
 				return
 			}
 
-			let objectOrUrl = typeof(obj) === "string" ? Global.appUrl(obj) : obj
+			let objectOrUrl = typeof(obj) === "string" ? ".." + obj : obj
```
Comment (jpetrell) [inline components/PageStack.qml:72]: If you make this relative did you test that the loading from qrc still works in non-file system build?

I would remove Global.appUrl(), it is now only used in one place in NavBar.qml.

**Comment by jpetrell** [comment]
Tested that I could modify MainView, PageStack, OverviewPage, SettingsPage, NavBar, InputPanel, PageSettingsWifi, PageSettingsGpsList. :partying_face:

File: components/PageStack.qml (hunk 2 above)
Comment (DanielMcInnes) [reply components/PageStack.qml:72]: Yes I did, it works ok. Removed Global.appUrl().

File: components/NavBar.qml [hunk 3]
```diff
@@ -108,7 +108,7 @@ Rectangle {  // Use an opaque background so that page disappears behind nav bar
 					height: width
 					radius: Theme.geometry_notificationsPage_delegate_marker_radius
 					color: Theme.color_critical
-					visible: model.url === Global.appUrl("/pages/NotificationsPage.qml")
+					visible: model.url === "qrc:/qt/qml/Victron/VenusOS/pages/NotificationsPage.qml"
```
Comment (chriadam) [inline components/NavBar.qml:111]: I'm confused about this.  What if it is loaded from the filesystem rather than qrc?

File: pages/MainView.qml [hunk 4]
```diff
@@ -88,7 +88,9 @@ Item {
 
 		z: 1
 		opacity: 0.0
-		source: Global.appUrl("/pages/ControlCardsPage.qml")
+		sourceComponent: Component {
```
Comment (chriadam) [inline pages/MainView.qml:91]: I think QML will automatically add Component { } sugar, so you can just do sourceComponent: ContolCardsPage {} for brevity, I believe.

**Comment by chriadam** [comment]
Typo in commit message (giles -> files)

**Comment by chriadam** [comment]
A couple of comments, but mostly LGTM

File: components/NavBar.qml (hunk 3 above)
Comment (DanielMcInnes) [reply components/NavBar.qml]: Whoops, thanks Chris

File: pages/MainView.qml (hunk 4 above)
Comment (DanielMcInnes) [reply pages/MainView.qml]: Done

File: ApplicationContent.qml [hunk 5]
```diff
@@ -88,18 +88,22 @@ Item {
...
+
+	Component {
+		id: inputPanelComponent
+
+		InputPanel {
+			id: inputPanel
+
+			mainViewItem: mainView
+			Component.onCompleted: Global.inputPanel = inputPanel
 		}
```
Comment (jpetrell) [inline ApplicationContent.qml:107]: Could just write 

```
Loader {
    onLoaded: Global.inputPanel = item
    active: Qt.platform.os !== "wasm"
    asynchronous: true
    sourceComponent: InputPanel {
        mainViewItem: mainView
    }
}
```

Comment (DanielMcInnes) [reply ApplicationContent.qml:99]: Done

**Review by jpetrell — APPROVED** [review]

**Pull request merged**